db/write_controller.cc

```diff
@@ -20,10 +21,11 @@ std::unique_ptr<WriteControllerToken> WriteController::GetStopToken() {
 
 std::unique_ptr<WriteControllerToken> WriteController::GetDelayToken(
     uint64_t write_rate) {
-  total_delayed_++;
-  // Reset counters.
-  last_refill_time_ = 0;
-  bytes_left_ = 0;
+  if (0 == total_delayed_++) {
+    // Starting delay, so reset counters.
+    next_refill_time_ = 0;
```

Comment:
Should next_refill_time be reset every time?  Here is my logic:
- When you start the first time, there is no time (t0) and no bytes left, so both are 0 and the delay is X
- If you start another delay at t1 with rate Y, then the write rate between t0 and t1 should be X but the write rate from t1 forward should be Y.

So ideally you would like to recalculate the number of bytes left on the second delay (but cannot since you do not have a SystemClock) and tell the "GetDelay" to recalculate on the next call as well

Reply:
I don't think it's worth trying to be that accurate about exactly when different rates are applied. Keep it simple(-ish).

Reply:
A comment about ignoring this condition might be useful for future reference.